src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -502,6 +502,8 @@ def validate_metadata(
         metadata=metadata,
         validation=validation,
         lockfile=lockfile,
+        is_old_style_tool_lockfile=lockfile.resolve_name not in python_setup.resolves,
```

Comment:
The changes in this file are just improvements to the error message generation. The proof is in the test, which hardly changed at all.